src/main/java/org/jenkinsci/plugins/github_branch_source/Connector.java

```diff
@@ -90,8 +93,8 @@
 public class Connector {
   private static final Logger LOGGER = Logger.getLogger(Connector.class.getName());
 
-  private static final Map<ConnectionId, GitHubConnection> connections = new HashMap<>();
-  private static final Map<GitHub, GitHubConnection> reverseLookup = new HashMap<>();
+  private static final Map<ConnectionId, GitHubConnection> connections = new ConcurrentHashMap<>();
+  private static final Map<GitHub, GitHubConnection> reverseLookup = new ConcurrentHashMap<>();
```

Comment:
@timja 
What if we used `ConnectionId` here instead of `GitHubConnection`?  It would me two lookups but it would mean there was no chance of `reverseLookup` accidentally holding on to a reference to a dead connection.

Reply:
```suggestion
  private static final Map<GitHub, ConnectionId> reverseLookup = new ConcurrentHashMap<>();
```

Reply:
sounds reasonable

Reply:
@timja 
You want to do it or should I?

Reply:
you go for it, i don't have time right now